diff --git a/model/backbone.py b/model/backbone.py
index 79cd61c..aa99661 100644
--- a/model/backbone.py
+++ b/model/backbone.py
@@ -343,7 +343,7 @@ class InceptionResNetV2(nn.Module):
         return x
 
 
-def inceptionresnetv2(num_classes=1001, pretrained='imagenet'):
+def inceptionresnetv2(num_classes=1001, pretrained='imagenet', weights_path=None):
     r"""InceptionResNetV2 model architecture from the
     `"InceptionV4, Inception-ResNet..." <https://arxiv.org/abs/1602.07261>`_ paper.
     """
@@ -357,7 +357,7 @@ def inceptionresnetv2(num_classes=1001, pretrained='imagenet'):
         # model.load_state_dict(model_zoo.load_url(settings['url']))
         # model.load_state_dict(torch.load('./model/inceptionresnetv2.pth'))
         
-        pretrained_dict = torch.load('./model/inceptionresnetv2.pth')
+        pretrained_dict = torch.load(weights_path)
         model_dict = model.state_dict()
 
         pretrained_dict = {k: v for k, v in pretrained_dict.items() if k in model_dict}
