diff --git a/UGCVQA_FR_model.py b/UGCVQA_FR_model.py
index 4740c3f..57718ee 100644
--- a/UGCVQA_FR_model.py
+++ b/UGCVQA_FR_model.py
@@ -33,9 +33,9 @@ class L2pooling(nn.Module):
         return (out+1e-12).sqrt()
 
 class ResNet50(torch.nn.Module):
-    def __init__(self):
+    def __init__(self, pretrained=True):
         super(ResNet50, self).__init__()
-        resnet_pretrained_features = nn.Sequential(*list(models.resnet50(pretrained=True).children())[:-2])
+        resnet_pretrained_features = nn.Sequential(*list(models.resnet50(pretrained=pretrained).children())[:-2])
         self.stage1 = torch.nn.Sequential()
         self.stage2 = torch.nn.Sequential()
         self.stage3 = torch.nn.Sequential()
